gradio/blocks.py

```diff
@@ -616,19 +616,37 @@ def preprocess_data(self, fn_index, raw_input, state):
             processed_input = raw_input
         return processed_input
 
-    async def call_function(self, fn_index, processed_input):
+    async def call_function(self, fn_index, processed_input, generator=None):
         """Calls and times function with given index and preprocessed input."""
         block_fn = self.fns[fn_index]
-
+        is_generating = False
         start = time.time()
+
         if inspect.iscoroutinefunction(block_fn.fn):
             prediction = await block_fn.fn(*processed_input)
         else:
             prediction = await anyio.to_thread.run_sync(
                 block_fn.fn, *processed_input, limiter=self.limiter
             )
+
+        if inspect.isgeneratorfunction(block_fn.fn):
+            try:
+                if generator is None:
+                    generator = prediction
+                prediction = next(generator)
+                is_generating = True
+            except StopIteration:
+                prediction = None
```

Comment:
I think this will fix the issue with functions yielding tuples erroring out

```suggestion
                n_outputs = len(self.dependencies[fn_index].get("outputs"))
                if n_outputs > 1:
                    stop_value = [None] * n_outputs
                else:
                    stop_value = None
                prediction = stop_value
```

Reply:
Nice catch! Thank you